src/layer/tile/GridLayer.js

```diff
@@ -431,12 +431,8 @@ L.GridLayer = L.Layer.extend({
 
 		if (center === undefined) { center = map.getCenter(); }
 		if (zoom === undefined) { zoom = map.getZoom(); }
-		var tileZoom = Math.round(zoom);
 
-		if (tileZoom > this.options.maxZoom ||
-			tileZoom < this.options.minZoom) { return; }
```

Comment:
Somehow I assumed that check was already wherever `this._tileZoom` was being calculated. Apparently it is not.